### PR TITLE
platform: Add `@ConditionalOnAnySepsEnabled` and `@ConditionalOnAllSepsEnabled` annotation 

### PR DESCRIPTION
**Pull request by lijamie98**
Added @ConditionalOnSepsEnabled annotation to enable or disable a Spring bean according to SEP enabled flag

<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add @ConditionalOnSepsEnabled annotation so that we may turn on/off Spring beans based on the SEPs enabled flag in the YAML file.

### Why

Dynamic module loading. 

### Known limitations
N/A

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/controller/Sep31Controller.java [hunk 1]
```diff
@@ -11,13 +11,16 @@
...
+import org.stellar.anchor.platform.condition.ConditionalOnSepsEnabled;
 import org.stellar.anchor.sep10.JwtToken;
 import org.stellar.anchor.sep31.Sep31Service;
 import org.stellar.anchor.sep31.Sep31Service.Sep31CustomerInfoNeededException;
 import org.stellar.anchor.sep31.Sep31Service.Sep31MissingFieldException;
 
 @RestController
+@CrossOrigin(origins = "*")
 @RequestMapping("sep31")
+@ConditionalOnSepsEnabled(seps = {"sep31"})
```
Comment (marcelosalloum) [inline platform/src/main/java/org/stellar/anchor/platform/controller/Sep31Controller.java:23]: Do you think SEPs 12 and 10 should both be mandatory for SEP-31? I'm not sure either.

**Review by marcelosalloum — COMMENTED** [review]
I absolutely love to see this landing ❤️!

I've left a comment about AND/OR logic when having a list of SEPs.

Also, it'd be great if we could make kafka conditional as well, so anchors can easily implement SEP-1, SEP-10 and SEP-12 without needing to provide a queue service. Maybe this is something for another pull request though.

File: platform/src/main/java/org/stellar/anchor/platform/condition/OnSepEnabledCondition.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      String[] seps = (String[]) annotationAttributes.get("seps");
+      for (String sep : seps) {
+        boolean enabled =
+            Boolean.parseBoolean(
+                context
+                    .getEnvironment()
+                    .getProperty(String.format("%s.enabled", sep.toLowerCase())));
+        if (enabled) return ConditionOutcome.match(String.format("%s enabled", className));
+      }
+    }
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/condition/OnSepEnabledCondition.java]: I had replaced the `@ConditionalOnSepsEnabled` with the following two annotations:

- `ConditionalOnAnySepsEnabled`
- `ConditionalOnAllSepsEnabled`

**Comment by lijamie98** [comment]
> I absolutely love to see this landing ❤️!
> 
> I've left a comment about AND/OR logic when having a list of SEPs.
> 
> Also, it'd be great if we could make kafka conditional as well, so anchors can easily implement SEP-1, SEP-10 and SEP-12 without needing to provide a queue service. Maybe this is something for another pull request though.

I had submit another PR to make Kafka conditional as well. 👍

**Review by marcelosalloum — APPROVED** [review]

**Pull request merged**